server/src/internalClusterTest/java/org/opensearch/search/simple/SimpleSearchIT.java

```diff
@@ -57,4 +321,379 @@ public void testSimpleTerminateAfterCount() throws Exception {
         assertHitCount(searchResponse, max);
         assertFalse(searchResponse.isTerminatedEarly());
     }
+
+    public void testSimpleTerminateAfterTrackTotalHitsUpTo() throws Exception {
```

Comment:
Added a new test case to cover all the different `terminate_after`/`track_total_hits` scenarios